Added support for LocaleId's, Added access members for fields and Significant refactoring

arjanmels opened the pull request:
Main changes:
- Include support for LocaleId's and printing of all fields (currency, script, variants, subdivisions)
- Add support for currency and subdivisions CLDR data
- Improved compile times, resulting file size etc. by changing maps from `CanonicalizedMap` to `const {}`, etc.
- Added non-countries to territories map
- Added static access members for languages, territories, scripts, variants and currencies.
- Many additional tests

I think the interface is backward compatible, hence a proposed version 2.1.0, but you may want to double check.

xvrh commented:
The removable of `CanonicalizedMap` means that the access are no longer case insensitive, right? If so, we may want to call it out in the changelog with an example of code before and after.

Otherwise that is good for me 👍

xvrh commented on .gitignore:
```diff
@@ -1,6 +1,7 @@
 # Files and directories created by pub
 .dart_tool/
 .packages
+pubspec.lock
```
Nice!

arjanmels commented:
> The removable of `CanonicalizedMap` means that the access are no longer case insensitive, right? If so, we may want to call it out in the changelog with an example of code before and after.
Via the ```operator []``` it is still case in-sensitive as I modified the operator to adjust the casing. Via direct access to ```language.languages```, ```territories.territories``` etc. it has indeed become case sensitive. 

I have added a note to CHANGELOG.md and updated the documentation of the individual classes.

arjanmels commented:
@xvrh Do you want me to try go ahead to merge and publish?

xvrh commented:
Yes that is good for me we can merge. Thanks a lot

Thanks for clarification about the case sensitivity.

arjanmels commented:
Hmm. The package has become too big before compression (~200MB and limit is 100MB), so cannot publish to pub.dev.
I'll look into solutions: 1) use typedefs to shorten the file content; 2) leave out some locales

xvrh commented:
@arjanmels , no problem to remove some locales 👍

arjanmels commented:
I found a better solution: inheritance for the country specific languages and only overriding what is different. It reduces the source code from ~200MB to ~75MB. Still need to do some cleanup, but mostly works.
